### PR TITLE
Add privilege management for account objects and schemas

### PR DESCRIPTION
**Pull request by kbial**
**What**
Grant/revoke privileges to roles on account objects (database, warehouse, resource monitor) and schemas. No updating possible.

**Why**
So we can grant privileges for account objects and schemas in Snowflake. 

We limited the functionality to only creating/deleting/reading the resource because any changes require deleting the current resource and creating a new one.

**Plans**
**Account object**
Create
```
------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  + snowflake_account_object_grant.example
      id:                    <computed>
      grant_option:          "false"
      object_name:           "MASTER"
      object_type:           "DATABASE"
      privileges.#:          "3"
      privileges.118361983:  "MODIFY"
      privileges.2914988887: "ALL"
      privileges.3729811359: "MONITOR"
      role:                  "EXAMPLE"


Plan: 1 to add, 0 to change, 0 to destroy.

------------------------------------------------------------------------
```
And that grants all the privileges to role EXAMPLE on database MASTER.

Delete
```
------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  - destroy

Terraform will perform the following actions:

  - snowflake_account_object_grant.example


Plan: 0 to add, 0 to change, 1 to destroy.

------------------------------------------------------------------------
```

**Schema**
Create
```
------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  + snowflake_schema_grant.example
      id:                    <computed>
      database:              "MASTER"
      grant_option:          "true"
      privileges.#:          "3"
      privileges.118361983:  "MODIFY"
      privileges.2914988887: "ALL"
      privileges.3729811359: "MONITOR"
      role:                  "kurczak"
      schema:                "TEST"


Plan: 1 to add, 0 to change, 0 to destroy.

------------------------------------------------------------------------
```

Delete
```

------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  - destroy

Terraform will perform the following actions:

  - snowflake_schema_grant.example


Plan: 0 to add, 0 to change, 1 to destroy.

------------------------------------------------------------------------
```

### CODE REVIEW

File: snowflake/resource_account_object_grant.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+
+	log.Println("Executing statement:", stmtSQL)
+	_, err := db.Exec(stmtSQL)
+	if err == nil {
+		d.SetId("")
+	}
+	return err
+}
+
+func getPrivilegesString(priviligesSet *schema.Set) string {
```
Comment (wookasz) [inline snowflake/resource_account_object_grant.go]: unused

File: snowflake/resource_schema_grant.go [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+		_ = txn.Rollback()
+	}()
+
+	var (
+		schemaName   = d.Get("schema").(string)
+		databaseName = d.Get("database").(string)
+		role         = d.Get("role").(string)
+	)
+
+	stmtSQL := fmt.Sprintf("USE DATABASE \"%s\"", databaseName)
```
Comment (wookasz) [inline snowflake/resource_schema_grant.go:69]: nit: I don't think you need the `USE DATABASE` you should be able to modify `generateRecipientSchemaString` to return `$database.$schema` instead

**Review by wookasz — APPROVED** [review]
- [x] can you add a note to the readme that you should not mix managing grants for a role with and without terraform, otherwise terraform may delete grants which were added outside of terraform when performing updates or deletes
- [x] couple comments
- [x] :shipit:

File: snowflake/resource_schema_grant.go (hunk 2 above)
Comment (kbial) [reply snowflake/resource_schema_grant.go:69]: I tried that before too but unfortunately this is the error I get: 
* snowflake_schema_grant.example: 090105 (22000): Cannot perform GRANT. This session does not have a current database. Call 'USE DATABASE', or use a qualified name.

**Pull request merged**